internal/envoy/v3/listener.go

```diff
@@ -389,6 +389,14 @@ func (b *httpConnectionManagerBuilder) Get() *envoy_listener_v3.Filter {
 		UseRemoteAddress: protobuf.Bool(true),
 		NormalizePath:    protobuf.Bool(true),
 
+		// We can ignore any port number supplied in the Host/:authority header
+		// before processing by filters or routing.
+		// Note that the port a listener is bound to will already be selected
+		// and that the port is stripped from the header sent upstream as well.
+		StripPortMode: &http.HttpConnectionManager_StripAnyHostPort{
```

Comment:
Note: I did not mention the redirect/multiple listener thing b/c I don't think it should be relevant.
- The hierarchy of the configs is Listener -> Filter chain -> Route
- When we build the redirect at the Route level, we know in contour what the "partner" secure port of the plaintext HTTP port should be and can set the redirect to the port with this field: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-redirectaction-port-redirect
- The redirect will result in a whole new request to a different Listener so stripping the port at the HTTP Connection Manager filter level should be irrelevant